### PR TITLE
125 input cursor jumps to end while typing

### PR DESCRIPTION
**Pull request by sudhacheran**
Fix for the cursor issue that jumps to the end while typing in the string input and number input components. 

To recreate the bug: In the Input field (String/Number input), where there is pre-existing text, attempt to add/update a character/number in the middle of the text. When a single character is typed, the cursor automatically moves to the end of the text/number.

Bug Fix:  Code changed to ensure that the cursor maintains its position where the user clicks on the text/number for editing, adding, or deleting characters/numbers, respectively.

- Changes were tested in Android (Samsung S20 - Chrome), iPad (Safari and Chrome) and MacBook (Chrome & Safari)

### CODE REVIEW

**Comment by sudhacheran** [comment]
- Recreating the bug:  In the Input field(String / Number input), where there is text already present and trying to add/update the text/number in the middle moves the cursor to the end of the text / number when single character is typed. 
- Fix:  Code change makes to cursor to maintain the position where the user clicks on the text/number for edit/add/delete respectively

- Changes were tested in Android (Samsung S20 - Chrome), iPad (Safari and Chrome) and MacBook (Chrome & Safari)

**Review by charliepnnl — APPROVED** [review]

File: src/components/number_input.tsx [hunk 1]
```diff
@@ -41,6 +42,16 @@ const NumberInput: FC<NumberInputProps> = ({
...
     const [error, setError] = useState<string>('')
 
+    // use the cursor postion when user edits the data in the component
+    const [cursor, setCursor] = useState<number | null>(null)
+    const ref = useRef<HTMLInputElement>(null)
+
+    // Refresh after the first render and every time the component updates
+    useEffect(() => {
+        const input = ref.current
+        if (!!input) input.setSelectionRange(cursor, cursor)
```
Comment (charliepnnl) [inline src/components/number_input.tsx:52]: My bad. Can you change `!!input` to `input`?  I put the !! in when I was trying to fix the Typescript error - it didn't help.

**Review by charliepnnl — APPROVED** [review]
Please remove my !! (see comment)
After that, please merge this into main.

File: src/components/number_input.tsx (hunk 1 above)
Comment (sudhacheran) [reply src/components/number_input.tsx:52]: Done

**Review by charliepnnl — APPROVED** [review]

**Review by charliepnnl — APPROVED** [review]

**Review by charliepnnl — APPROVED** [review]

**Comment by charliepnnl** [comment]
GitHub seems to not be recognizing my approval. So I will bypass the review requirement and merge it in myself.

**Pull request merged**